src/plugins/intel_cpu/src/nodes/rdft.cpp

```diff
@@ -0,0 +1,1537 @@
+    if (!data_precision.is_float()) {
+        IE_THROW() << error_msg_prefix << " has unsupported 'data' input precision: " << data_precision.name();
+    }
+
+    const auto& axes_precision = getOriginalInputPrecisionAtPort(AXES_INDEX);
+    if (axes_precision != Precision::I32 && axes_precision != Precision::I64) {
+        IE_THROW() << error_msg_prefix << " has unsupported 'axes' input precision: " << axes_precision.name();
+    }
+
+    if (inputShapes.size() > SIGNAL_SIZE_INDEX) {
+        const auto& signal_size_precision = getOriginalInputPrecisionAtPort(SIGNAL_SIZE_INDEX);
+        if (signal_size_precision != Precision::I32 && signal_size_precision != Precision::I64) {
+            IE_THROW() << error_msg_prefix << " has unsupported 'signal_size' input precision: " << signal_size_precision.name();
+        }
+    }
+
+    std::vector<PortConfigurator> configurators({{LayoutType::ncsp, Precision::FP32},
+                                                 {LayoutType::ncsp, Precision::I32}});
+    if (inputShapes.size() > SIGNAL_SIZE_INDEX)
+        configurators.push_back({LayoutType::ncsp, Precision::I32});
+
+    addSupportedPrimDesc(configurators, {{LayoutType::ncsp, Precision::FP32}}, impl_desc_type::ref_any);
+}
+
+void RDFT::execute(dnnl::stream strm) {
+    const auto& input_mem = getParentEdgeAt(DATA_INDEX)->getMemory();
+    const auto& output_mem = getChildEdgeAt(0)->getMemory();
+    const auto& input_shape = input_mem.getStaticDims();
+    const auto& output_shape = output_mem.getStaticDims();
+
+    auto input_ptr = reinterpret_cast<float*>(input_mem.GetPtr());
+    auto output_ptr = reinterpret_cast<float*>(output_mem.GetPtr());
+
+    auto rank = inverse ? output_shape.size() : input_shape.size();
+    const auto& axes_mem = getParentEdgeAt(AXES_INDEX)->getMemoryPtr();
+    auto axes_ptr = reinterpret_cast<const int32_t*>(axes_mem->GetPtr());
+    std::vector<int> axes(axes_ptr, axes_ptr + axes_mem->getStaticDims()[0]);
+    normalize_axes(axes, rank, inverse);
+
+    const auto& input_strides = input_mem.GetDescWithType<BlockedMemoryDesc>()->getStrides();
+    const auto& output_strides = output_mem.GetDescWithType<BlockedMemoryDesc>()->getStrides();
+
+    executor->execute(input_ptr, output_ptr, rank, axes,
+                      input_shape, output_shape,
+                      input_strides, output_strides);
+}
+
+bool RDFT::created() const {
+    return getType() == Type::RDFT;
+}
+
+void RDFT::createPrimitive() {
+    if (inputShapesDefined()) {
+        if (needPrepareParams())
+            prepareParams();
+        updateLastInputDims();
+    }
+}
+
+bool RDFT::needPrepareParams() const {
+    return true;
+}
+
+
+namespace {
+template <typename T>
+size_t complex_type_size() {
+    return sizeof(T) * 2;
+}
+
+enum dft_type {
+    real_to_complex,
+    complex_to_complex,
+    complex_to_real,
+};
+
+#define GET_OFF(field) offsetof(jit_dft_args, field)
```

Comment:
Please move all JIT related code to separate file in folder intel_cpu/src/nodes/kernels